### PR TITLE
Migrate db

### PR DESCRIPTION
**Pull request by LikeMyBread**
This is still relatively broken, but it includes some fixes such as the ability to log out.

### CODE REVIEW

File: saylua/templates/navbar.html [hunk 1]
```diff
@@ -97,7 +97,7 @@
       </a>
       <div id="dropdown-user-menu" class="dropdown-menu">
         <div id="user-menu" class="menu">
-          <h3>Logged in as {{g.user.display_name}}</h3>
+          <h3>Logged in as {{g.user.display_name.display_name}}</h3>
```
Comment (NoiSek) [inline saylua/templates/navbar.html]: We've really got to change this before it's too far gone, haha. `display_name.name` or `display_name.username` would be way better.

This is unrelated to your changes, don't feel obligated to fix this.

**Review by NoiSek — APPROVED** [review]
Looks good, one step closer to the future.

File: saylua/modules/messages/models/db.py [hunk 2]
```diff
@@ -1,6 +1,45 @@
...
 import datetime
+from saylua import db
+
+
+class Conversation(db.Model):
+    __tablename__ = "conversations"
+
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(256))
+    author = db.Column(db.Integer, db.ForeignKey('users.id'))
```
Comment (tiffz) [inline saylua/modules/messages/models/db.py]: Remove.

File: saylua/modules/messages/models/db.py [hunk 3]
```diff
@@ -1,6 +1,45 @@
...
+
+
+class ConversationMember(db.Model):
+    __tablename__ = "conversation_members"
+
+    conversation_id = db.Column(db.Integer, db.ForeignKey('conversations.id'), primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), primary_key=True)
+    unread = db.Column(db.Boolean, default=False)
+
```
Comment (tiffz) [inline saylua/modules/messages/models/db.py]: Given the way messages are used, I think it would be best to denormalize the title from Conversation into this model, so we don't have to do so many join queries everywhere when looking up someone's messages. 

Also, add a last updated time for this field for similar reasons. Since messages are displayed on every page, it's worth it to have a little bit of denormalization, I think.

File: saylua/modules/messages/models/db.py [hunk 4]
```diff
@@ -1,6 +1,45 @@
...
+
+class Message(db.Model):
+    __tablename__ = "messages"
+
+    id = db.Column(db.Integer, primary_key=True)
+    conversation_id = db.Column(db.Integer, db.ForeignKey('conversations.id'))
+    author = db.Column(db.Integer, db.ForeignKey('users.id'))
```
Comment (tiffz) [inline saylua/modules/messages/models/db.py]: author -> author_id

File: saylua/modules/messages/models/db.py (hunk 3 above)
Comment (tiffz) [inline saylua/modules/messages/models/db.py]: Maybe we could add like a .name() function to the user model that was a wrapper around this? Agree with changing display_name.display_name regardless haha.
